### PR TITLE
feat(clipboard): support `readImage` & `writeImage`

### PR DESCRIPTION
**Pull request by cijiugechu**
related issue:
- [[feat] Clipboard support for RTF (rich text format), HTML and Image](https://github.com/tauri-apps/tauri/issues/7750)

### CODE REVIEW

File: plugins/clipboard-manager/src/mobile.rs
Comment (amrbashir) [inline plugins/clipboard-manager/src/mobile.rs]: Return an Error instead of panic

File: plugins/clipboard-manager/src/commands.rs
Comment (amrbashir) [inline plugins/clipboard-manager/src/commands.rs:28]: let's also change `write` command to `write_text`

File: plugins/clipboard-manager/src/desktop.rs
Comment (amrbashir) [inline plugins/clipboard-manager/src/desktop.rs]: let's remove this and only expose `write_text` and `write_image` variants.

File: plugins/clipboard-manager/src/desktop.rs
Comment (amrbashir) [inline plugins/clipboard-manager/src/desktop.rs]: let's remove this as well

File: plugins/clipboard-manager/Cargo.toml
Comment (amrbashir) [inline plugins/clipboard-manager/Cargo.toml:28]: I am not too happy with the current state of images handling in tauri overall and before we merge this PR I think it would be much better to have a better representation in tauri core similar to electron's `NativeImage` class. 

cc @lucasfernog

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by simonhyll — APPROVED** [review]

File: plugins/clipboard-manager/Cargo.toml
Comment (FabianLars) [reply plugins/clipboard-manager/Cargo.toml:28]: isn't the image crate basically the ecosystem default for that? I'd much rather agree on one of its types (like RgbaImage or DynamicImage idk) across all of our crates than creating our own type(s) (again).

Comment (FabianLars) [reply plugins/clipboard-manager/Cargo.toml:28]: Also something i don't get about arboard, they already depend on the image crate but still created their own image type. imho super annoying...

Comment (amrbashir) [reply plugins/clipboard-manager/Cargo.toml:28]: > isn't the image crate basically the ecosystem default for that? I'd much rather agree on one of its types (like RgbaImage or DynamicImage idk) across all of our crates than creating our own type(s) (again).

yes but image crate is a heavy dependency and you may not need most of it, for that reason, crates like `winit`, `muda` and `tray-icon` opted into a light icon type.

What I meant by my previous comment is that we need something like `NativeImage` that is light like `winit`'s `Icon` type, with a few helper methods to support creating from:
- png / jpeg / ico bytes (based on feature flags enabled)
- png / jpeg / ico files (based on feature flags enabled)
- from internal embedded assets (if possible, not sure how yet)
- raw rbga8
- can have a lighter representation in JS (using tauri's new resource table) so we don't need to send bytes forth and back between JS and Rust (unless user wants to get it in JS)
- unlike the existing `tauri::Icon`, it will be a struct not an Enum

Comment (FabianLars) [reply plugins/clipboard-manager/Cargo.toml:28]: Sounds like the helper functions would negate most of the "lightness" compared to image::RgbaImage honestly (with its default features disabled of course) but i don't have any hard feelings about it, as long as we have _one_ type we can re-use across all repos 🤷

Comment (amrbashir) [reply plugins/clipboard-manager/Cargo.toml:28]: Not really, the helper functions will use the lighter crates `png` and `ico` and will be behind a feature flag as well same as it is now in v1.

**Comment by lucasfernog** [comment]
I wish we had https://github.com/1Password/arboard/pull/103 :(

**Review by lucasfernog — DISMISSED** [review]

File: plugins/authenticator/src/u2f_crate/protocol.rs
Comment (amrbashir) [inline plugins/authenticator/src/u2f_crate/protocol.rs:99]:
```suggestion
        // Safe to unwrap since 300 is within the constraints of Duration::try_seconds
        if expiration(challenge.timestamp) > Duration::try_seconds(300).unwrap() {
```

File: plugins/authenticator/src/u2f_crate/protocol.rs
Comment (amrbashir) [inline plugins/authenticator/src/u2f_crate/protocol.rs:157]:
```suggestion
        // Safe to unwrap since 300 is within the constraints of Duration::try_seconds
        if expiration(challenge.timestamp) > Duration::try_seconds(300).unwrap() {
```

File: plugins/clipboard-manager/Cargo.toml
Comment (amrbashir) [inline plugins/clipboard-manager/Cargo.toml:28]:
```suggestion
```

File: plugins/clipboard-manager/guest-js/index.ts
Comment (amrbashir) [inline plugins/clipboard-manager/guest-js/index.ts]: this should return an instance of `Image` from `@tauri-apps/image`

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Comment by lucasfernog** [comment]
Just waiting on the next tauri release to push the image changes :)

File: plugins/clipboard-manager/guest-js/index.ts
Comment (lucasfernog) [reply plugins/clipboard-manager/guest-js/index.ts]: in this case we already have the bytes/width/height and the Image class does note expose an API to directly create an image from those values.. doesn't make much sense to send them back to the rust side...

Comment (amrbashir) [reply plugins/clipboard-manager/guest-js/index.ts]: The `Image` class should be constructed on the Rust side then added to the resources_table, we only return an RresourceId to JS and construct the Js Image class with it.

**Pull request merged**
